src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -126,9 +127,17 @@ internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
             {
                 fileLocation.TryReconstructAbsoluteUri(_run.OriginalUriBaseIds, out Uri uri);
 
-                // TODO: we should convert node.Encoding to a .NET equivalent and pass it here
-                // https://github.com/Microsoft/sarif-sdk/issues/934
-                node = FileData.Create(uri, _dataToInsert, node.MimeType, encoding: null);
+                Encoding encoding = null;
+
+                try
+                {
+                    encoding = Encoding.GetEncoding(node.Encoding);
+                }
+                catch (ArgumentException) { }
+
+                int length = node.Length;
+                node = FileData.Create(uri, _dataToInsert, node.MimeType, encoding: encoding);
+                node.Length = length;
```

Comment:
>length [](start = 30, length = 6)

add simple unit tests for the nice changes you have made. they don't need to be complex or super comprehensive, just covering your code delta.

Reply:
this change is approved, check in when you fix the suggestions above. 

---
In reply to: [208705942](https://github.com/Microsoft/sarif-sdk/pull/998#discussion_r208705942) [](ancestors = 208705942)